angular/projects/researchdatabox/form/src/app/component/textarea.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, Input } from '@angular/core';
+import { FormFieldBaseComponent, FormFieldCompMapEntry, FormFieldModel } from "@researchdatabox/portal-ng-common";
+import { TextareaComponentConfig } from '@researchdatabox/sails-ng-common/dist/src/config/component/textarea.model';
+import { get as _get, isUndefined as _isUndefined, isEmpty as _isEmpty } from 'lodash-es';
+
+export class TextareaModel extends FormFieldModel<string> {
+}
+
+@Component({
+  selector: 'redbox-textarea',
+  template: `
+    @if (isVisible) {
+      <ng-container *ngTemplateOutlet="getTemplateRef('before')" />
+      <textarea [formControl]="formControl"
+        class="form-control"
+        [class.is-valid]="isValid"
+        [class.is-invalid]="!isValid"
+        [attr.required]="isRequired === true ? true : null"
+        [attr.disabled]="isDisabled ? 'true' : null"
+        [attr.readonly]="isReadonly ? 'true' : null"
+        [attr.title]="tooltip ? tooltip : tooltipPlaceholder"
+        [attr.rows]="rows"
+        [attr.cols]="cols"
+        [attr.placeholder]="placeholder"></textarea>
+      <ng-container *ngTemplateOutlet="getTemplateRef('after')" />
+    }
+  `,
+  standalone: false
+})
+export class TextareaComponent extends FormFieldBaseComponent<string> {
+  protected override logName: string = "TextareaComponent";
+  public tooltip:string = '';
+  public tooltipPlaceholder:string = 'placeholder';
+  public rows:number = 0;
+  public cols:number = 0;
+  public placeholder: string | undefined = '';
+
+  /**
+   * Override to set additional properties required by the wrapper component.
+   *
+   * @param formFieldCompMapEntry
+   */
+  protected override setPropertiesFromComponentMapEntry(formFieldCompMapEntry: FormFieldCompMapEntry): void {
+    super.setPropertiesFromComponentMapEntry(formFieldCompMapEntry);
+    this.tooltip = this.getTooltip();
+    this.tooltipPlaceholder = '';
+    let textareaConfig = this.componentDefinition?.config as TextareaComponentConfig;
+    if(!_isUndefined(textareaConfig) && _isEmpty(textareaConfig)) {
+      let defaultConfig = new TextareaComponentConfig();
+      this.rows = textareaConfig.rows || defaultConfig.rows;
+      this.cols = textareaConfig.cols || defaultConfig.cols;
```

Comment:
textareaConfig.rows and textareaConfig.cols is always going to have no value in this context. I think it makes sense to invert the logic a bit. Something like

const textareaConfig = this.componentDefinition?.config as TextareaComponentConfig | undefined;
const defaultConfig = new TextareaComponentConfig();
const cfg = (_isUndefined(textareaConfig) || _isEmpty(textareaConfig)) ? defaultConfig : textareaConfig;
this.rows = cfg.rows ?? defaultConfig.rows;
this.cols = cfg.cols ?? defaultConfig.cols;
this.placeholder = cfg.placeholder ?? defaultConfig.placeholder;

That way you get the default rows and cols regardless of whether the entire text area config is empty.